CMakeLists.txt

```diff
@@ -1,141 +1,32 @@
 
-    include/nil/crypto3/zk/components/hashes/knapsack/r1cs/knapsack.hpp
-    include/nil/crypto3/zk/components/hashes/sha256/r1cs/sha256_aux.hpp
-    include/nil/crypto3/zk/components/hashes/sha256/r1cs/sha256_components.hpp
-    include/nil/crypto3/zk/components/hashes/sha256/r1cs/sha256_component.hpp
-
-    include/nil/crypto3/zk/components/merkle_tree/r1cs/authentication_path.hpp
-    include/nil/crypto3/zk/components/merkle_tree/r1cs/check_read.hpp
-    include/nil/crypto3/zk/components/merkle_tree/r1cs/check_update.hpp
-
-    include/nil/crypto3/zk/components/algebra/pairing/detail/r1cs/mnt4.hpp
-    include/nil/crypto3/zk/components/algebra/pairing/detail/r1cs/mnt6.hpp
-    include/nil/crypto3/zk/components/algebra/pairing/weierstrass/r1cs/pairing_checks.hpp
-    include/nil/crypto3/zk/components/algebra/pairing/weierstrass/r1cs/final_exponentiation.hpp
-    include/nil/crypto3/zk/components/algebra/pairing/weierstrass/r1cs/miller_loop.hpp
-    include/nil/crypto3/zk/components/algebra/pairing/weierstrass/r1cs/precomputation.hpp
-
-    include/nil/crypto3/zk/components/systems/snark/r1cs_pp_zksnark/verifier.hpp
-    include/nil/crypto3/zk/components/detail/r1cs/packing.hpp
-    include/nil/crypto3/zk/components/boolean/r1cs/comparison.hpp
-    include/nil/crypto3/zk/components/boolean/r1cs/conjunction.hpp
-    include/nil/crypto3/zk/components/boolean/r1cs/disjunction.hpp
-    include/nil/crypto3/zk/components/boolean/r1cs/inner_product.hpp
-    include/nil/crypto3/zk/components/detail/r1cs/loose_multiplexing.hpp
-    include/nil/crypto3/zk/components/component_from_r1cs.hpp
-    include/nil/crypto3/zk/component.hpp
-    include/nil/crypto3/zk/blueprint/r1cs.hpp
-    include/nil/crypto3/zk/blueprint/plonk.hpp)
-
-list(APPEND ${CURRENT_PROJECT_NAME}_UNGROUPED_SOURCES)
-
-list(APPEND ${CURRENT_PROJECT_NAME}_HEADERS ${${CURRENT_PROJECT_NAME}_PUBLIC_HEADERS})
-
-list(APPEND ${CURRENT_PROJECT_NAME}_SOURCES ${${CURRENT_PROJECT_NAME}_UNGROUPED_SOURCES})
-
-cm_setup_version(VERSION 0.1.0 PREFIX ${CMAKE_WORKSPACE_NAME}_${CURRENT_PROJECT_NAME})
-
-add_library(${CMAKE_WORKSPACE_NAME}_${CURRENT_PROJECT_NAME} INTERFACE)
-
-set_target_properties(${CMAKE_WORKSPACE_NAME}_${CURRENT_PROJECT_NAME} PROPERTIES
-                      EXPORT_NAME ${CURRENT_PROJECT_NAME})
-
-target_include_directories(${CMAKE_WORKSPACE_NAME}_${CURRENT_PROJECT_NAME} INTERFACE
+include(GNUInstallDirs)
+target_include_directories(blueprint INTERFACE
                            "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-                           "$<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>"
-
-                           ${Boost_INCLUDE_DIRS})
-
-target_link_libraries(${CMAKE_WORKSPACE_NAME}_${CURRENT_PROJECT_NAME} INTERFACE
+                           "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
+)
 
+target_link_libraries(blueprint INTERFACE
                       ${Boost_LIBRARIES}
                       crypto3::all
```

Comment:
Please remove crypto3::all as this breaks my non-flake development environment; and the flake redirects do not work due to cmake errrors.
This makes doing changes to both crypto3 and blueprint at the same time very difficult.

Reply:
Maybe this is possible to fix some other way?
I am unsure if anyone else is testing the 'intended' flake development way; but it does not seem to function for me.

Reply:
Seems like we agreed to use nix flakes as a common way of managing dependencies, you can see it already in master. I know it could be painful to switch to it :( 
You can try to run cmake inside `nix develop --override-input nil_crypto3 /path/to/crypto3 .?submodules=1#` shell. It should work now if your local crypto3 is on the current master.

Reply:
Now the cmake no longer fails, but it still doesn't work, the local changes are not actually propagated.
e.g. adding
```
should break the build
```
line (with no quotes) to one of my c++ includes inside crypto3 should break the build in blueprint.
But it doesn't; changes are not propagated. I have to run a new `nix develop` command every time I make a change to my other repository.
While developing zkevm I have to juggle both crypto3, blueprint, the actual zkevm repository and the assigner repositories, re-running `nix develop` every time something changes somewhere for each repository.
This is not fun.

Reply:
@Iluvmagick actually it is still possible to build any project without Nix. But you have to install all the dependencies and rebuild them when needed. Here is an example of a script that may be helpful:
```bash
#!/bin/bash -ex

# CRYPTO3_SRC=/path/to/crypto3
# CM_SRC=/path/to/cmake/modules
# CMAKE_PREFIX_PATH=/desired/install/path/

if [[ ! -d $CRYPTO3_SRC ]]; then
    echo "CRYPTO3_SRC = $CRYPTO_SRC is not a directory"
    exit 1
fi

if [[ ! -d $CM_SRC ]]; then
    echo "CM_SRC = $CM_SRC is not a directory"
    exit 1
fi

if [[ ! -d $CM_SRC ]]; then
    echo "CM_SRC = $CM_SRC is not a directory"
    exit 1
fi

if [[ -z $CMAKE_PREFIX_PATH ]]; then
    echo "CMAKE_PREFIX_PATH is not set"
    exit 1
fi

build_module () {
    MODULE_SRC=$1
    INTERNAL_PREFIX=$2
    cmake $MODULE_SRC -B $MODULE_SRC/build -DCMAKE_INSTALL_PREFIX=$CMAKE_PREFIX_PATH/$INTERNAL_PREFIX
    make -C $MODULE_SRC/build install
}

mkdir -p $CMAKE_PREFIX_PATH
build_module $CM_SRC ""
build_module $CRYPTO3_SRC crypto3
```
Usage:
```
export CMAKE_PREFIX_PATH=/some/desired/path/for/library/installation
CRYPTO3_SRC=/path/to/crypto3 CM_SRC=/path/to/cmake/modules script.sh
cd blueprint
cmake -B build ... # blueprint options
```
But of course if you change something in crypto3, you need to rerun `make -C crypto3/build install` to update installed headers.

Reply:
This solution seems to be the same as rerunning `nix develop` every time crypto3 changes, which is what I am already doing. I am not averse to using nix — the tests I've build for blueprint did run on a flake inside `nix develop` — I am averse to having to run commands every time I change something.

[This](https://github.com/NilFoundation/research-excalibur-visualizer/tree/78c1f92d28129d3361fbba4cec2382f589dc74f0) (commit with the version of crypto3 where changes to cmake files haven't broken the process yet) used to work for me.
I can just change the file inside crypto3, and next time I am building a test the changes are reflected in the build with zero additional work required.

Reply:
IIRC changing the `crypto3::all` back to the list of modules was basically all I had to do to make the environment functional again, besides adding some boost modules (but that could be done on the level above).
But need to check again, maybe the new cmake changes did make that part unnecessary?